### ISSUE TITLE
new-parens rule has false negative

### ISSUE THREAD

**Issue by mysticatea** [eslint/eslint#6997]
**What version of ESLint are you using?**

- 3.4.0

**What parser (default, Babel-ESLint, etc.) are you using?**

- default

**Please show your full configuration:**

- nothing

**What did you do? Please include the actual source code causing the issue.**

    $ echo "new (Foo)" | eslint --stdin --rule new-parens:error --no-eslintrc

**What did you expect to happen?**

```
<text>
  1:1  error  Missing '()' invoking a constructor  new-parens

✖ 1 problem (1 error, 0 warnings)
```

**What actually happened? Please include the actual, raw output from ESLint.**

No errors.
